extensions/schema-compare/src/test/schemaCompare.test.ts

```diff
@@ -178,8 +178,70 @@ describe('SchemaCompareMainWindow.execute', function (): void {
 		await result.execute();
 
 		//Generate script button and apply button should be enabled for database comparison
-		should(result.verifyButtonsState(true, true)).equal(true);
+		should(result.verifyButtonsState(true, true, true, true, true, false, true, true, true, true)).equal(true);
 	});
 
 });
 
+describe('SchemaCompareMainWindow.updateSourceAndTarget', function (): void {
+	before(() => {
+		mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
+		mockExtensionContext.setup(x => x.extensionPath).returns(() => '');
+		testContext = createContext();
+	});
+
+	it('Should set buttons appropriately when source and target endpoints are empty', async function (): Promise<void> {
+		let sc = new SchemaCompareTestService();
+		let endpointInfo: mssql.SchemaCompareEndpointInfo;
+
+		let result = new SchemaCompareMainWindow(testContext.apiWrapper.object, sc, mockExtensionContext.object);
+
+		await result.start(null);
+
+		should(result.getComparisonResult() === undefined);
+
+		result.sourceEndpointInfo = {...endpointInfo};
+		result.targetEndpointInfo = {...endpointInfo};
+
+		result.updateSourceAndTarget();
+
+		should(result.verifyButtonsState(false, false, false, true, false, false, true, true, false, false)).equal(true);
+	});
+
+	it('Should set buttons appropriately when source endpoint is empty and target endpoint is populated', async function (): Promise<void> {
+		let sc = new SchemaCompareTestService();
+		let endpointInfo: mssql.SchemaCompareEndpointInfo;
+
+		let result = new SchemaCompareMainWindow(testContext.apiWrapper.object, sc, mockExtensionContext.object);
+
+		await result.start(null);
+
+		should(result.getComparisonResult() === undefined);
+
+		result.sourceEndpointInfo = {...endpointInfo};
+		result.targetEndpointInfo = await setDacpacEndpointInfo(mocktarget);
+
+		result.updateSourceAndTarget();
+
+		should(result.verifyButtonsState(false, false, true, true, false, false, true, true, false, false)).equal(true);
+	});
+
+	it('Should set buttons appropriately when source and target endpoints are populated', async function (): Promise<void> {
```

Comment:
are there tests to validate the value of source and target endpoints?

Reply:
Yes those tests were added in a previous PR for utils.ts where getEndpointName is thoroughly tested.